wagtail/fields.py

```diff
@@ -117,25 +116,20 @@ def json_field(self):
         return models.JSONField(encoder=DjangoJSONEncoder)
 
     def _check_json_field(self):
-        if type(self.use_json_field) is not bool:
-            warnings.warn(
-                f"StreamField must explicitly set use_json_field argument to True/False instead of {self.use_json_field}.",
-                RemovedInWagtail50Warning,
-                stacklevel=3,
+        if self.use_json_field is not True:
+            # RemovedInWagtail60Warning - make use_json_field optional and default to True
+            raise ImproperlyConfigured(
+                "StreamField must explicitly set use_json_field=True"
```

Comment:
This looks good to me! Shall we update the docs in this PR as well, or is that going to be separate?

Reply:
Yes, we should. And on searching the docs for references to `use_json_field`, I realise that the recipe at https://docs.wagtail.org/en/latest/advanced_topics/streamfield_migrations.html#migrating-richtextfields-to-streamfield is no longer valid, and will need a total rewrite. Hmm...

Reply:
@laymonage Have now updated the richtext -> streamfield recipe to not depend on `use_json_field=False`, along with the other required doc updates.

I've realised that occurrences of the _old_ version of this snippet in people's migration histories are going to be a potential minefield: anyone setting up a new instance of a legacy site and running the migrations from scratch will have any StreamFields created as JSONField from the outset. And dropping in the new version may be non-trivial, if there are other migrations after it that must be run before the PageRevision->Revision change...

However, it turns out that the full revision-handling version of this snippet is already broken in 4.0, due to `page.revisions` being a property provided by `RevisionMixin` rather than a true relation (and thus unavailable in the frozen ORM). Given that apparently nobody has reported this, I'm inclined to believe that either people aren't using this snippet at all, or they have found their own workarounds, most likely "all the instances with data we care about were already migrated a year ago, so we can blank out this data migration with no ill effects" :-)

Reply:
Yep, sorry about that, I didn't have enough domain knowledge to update the migration examples when I did the generic `Revision` change. And yeah, given that we haven't got any reports about the broken example, I think your reasoning makes sense.Thanks!